src/components/PanelV2/PanelV2.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * @file Panel v2.
+ * @copyright IBM Security 2019
+ */
+
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { Component, createRef } from 'react';
+import Close20 from '@carbon/icons-react/lib/close/20';
+
+import { getComponentNamespace } from '../../globals/namespace';
+import * as defaultLabels from '../../globals/nls';
+
+import { isClient } from '../../globals/utils/capabilities';
+
+import IconButton from '../IconButton';
+import Transition from '../Transition';
+import Portal, { PORTAL_EVENTS } from '../Portal';
+
+export const namespace = getComponentNamespace('panelv2');
+
+/**
+ * Panel v2 container component.
+ * @param {Object.<string, *>} props Panel v2 container props.
+ * @returns {PanelV2} Panel v2 container instance.
+ */
+export default class PanelV2 extends Component {
+  state = { bodyMargin: 0, isOpen: this.props.isOpen };
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (prevState.isOpen !== nextProps.isOpen) {
+      return { isOpen: nextProps.isOpen };
+    }
+    return null;
+  }
+
+  componentDidMount() {
+    if (isClient() && this.state.isOpen) {
+      this.setBodyMargin();
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.isOpen && !prevState.isOpen && isClient()) {
```

Comment:
Looks good, this is the only line i'm worried about. 

Checking to make sure `prevState.isOpen` is false - will this not prevent consumers from from swapping out the panel content? 

E.g
- Panel children is swapped out to fill a larger space
- Because the panel was not explicitly closed, `isOpen` will be true
- `setBodyMargin()` will not be called in this instance.

Would it be possible to remove this condition from the check?

In the `qradar-ui` rewrite we use a single Panel component across our app, which simply swaps out the content when the user opens a new panel.

This is to prevent overlapping panel instances.

One defect we are seeing with the v1 panel is that if a user opens a v4 IP. `setBodyMargin` is called initially which leaves just enough margin for the v4 IP address:

But if they then open a v6 IP Address - `setBodyMargin` is not called and the following is seen:
![image](https://user-images.githubusercontent.com/22816887/63505004-a019e400-c4ca-11e9-96ef-06144849f5a2.png)

Reply:
Thank you @danielcooke1996! I agree with the premise here, but unfortunately removing that condition will cause a loop for `componentDidUpdate` & throw a "maximum depth" error 😭 

Generally though, I think we could still address your issue if we were to refactor the use of these lifecycle methods a bit. We currently have the older, deprecated methods anyway so there's some cleanup to be done. 🤔 

I think for now though... I'd like to keep things as-is just so we can get this experimental component in, for a couple teams to use. And we could address your concern (along with a lifecycle refactor) in a separate PR.